From 08f4f9367db2594350658235969d25c85c212c82 Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Thu, 16 Oct 2014 22:17:53 +0800
Subject: [PATCH 0137/5242] MLK-9617-16 usb: common: otg: clear a_conn flag
 when leaves b host

commit  b604d86a2c332b16b9774018ad2258edc715f9d9 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch is clearing a_conn flag when b device leave b_host state.

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit e511cc08879fa97035900ae8626051e473f30cd9)
(cherry picked from commit e293fb467244e7068bd67e6c37a485b99944bbcf)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/common/usb-otg-fsm.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index b6d04ea..9cbd655 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -83,6 +83,7 @@ static void otg_leave_state(struct otg_fsm *fsm, enum usb_otg_state old_state)
 			fsm->otg_hnp_reqd = 0;
 			fsm->b_bus_req = 0;
 		}
+		fsm->a_conn = 0;
 		break;
 	case OTG_STATE_A_IDLE:
 		fsm->adp_prb = 0;
-- 
1.7.9.5

